lib/ex_machina.ex

```diff
@@ -141,21 +141,24 @@ defmodule ExMachina do
   """
   def build(module, factory_name, attrs \\ %{}) do
     attrs = Enum.into(attrs, %{})
-    function_name = Atom.to_string(factory_name) <> "_factory" |> String.to_atom
+    function_name = build_function_name(factory_name)
     if Code.ensure_loaded?(module) && function_exported?(module, function_name, 0) do
       apply(module, function_name, []) |> do_merge(attrs)
     else
       raise UndefinedFactoryError, factory_name
     end
   end
 
-  defp do_merge(%{__struct__: _} = record, attrs) do
-    struct!(record, attrs)
-  end
-  defp do_merge(record, attrs) do
-    Map.merge(record, attrs)
+  defp build_function_name(factory_name) do
+    factory_name
+    |> Atom.to_string
+    <> "_factory"
```

Comment:
Oh, good catch. Running this code in my `iex` and seeing that it doesn't actually work...